src/core/qgsaggregatecalculator.h

```diff
@@ -155,10 +156,12 @@ class CORE_EXPORT QgsAggregateCalculator
      * If an expression is used, then the context parameter must be set.
      * \param context expression context for evaluating expressions
      * \param ok if specified, will be set to TRUE if aggregate calculation was successful
+     * \param ids List of feature ID to filter out unwanted features in the request
```

Comment:
This API is potentially tricky - it's impossible to differentiate if an empty list means "no filtering wanted" or was the result of some earlier filter which resulted in no matching features. I think it's better to make this argument a pointer which defaults to nullptr, and use nullptr as the indicator that no filtering is wanted.

Reply:
I made it this way since the request only accepts a list, and I didn't want to convert the pointer to a list. I agree that an empty list is ambiguous but this was the easiest way to deal with thins since the context scope provides a list and not a pointer to a list, in my main PR the list of features per a given symbol is provided by the expression context. This is why I did not go for a pointer. ( But I must still solve the damn segfaults).

Reply:
Yes, but this API could accept a pointer, and you just pass a reference to it when calling from your code (prefix it with "&")

Reply:
(and you dereference the pointer for the request by prefixing with a "*")

Reply:
True, do you want me to change it?
In my case it would mean going from a const QgsFeatureIds to a pointer to a const QgsFeatureIds back to a constQgsFeatureIds.

If you tell me this is fine I'll change it right away.

Reply:
Yes, I think an explicit API is safer than one which can be interpreted two ways (and would likely lead to future bugs as a result).

Don't forget to add unit tests for this too!